From de0eca0d1d2b39eba07e365497e67edcab44a7ce Mon Sep 17 00:00:00 2001
From: Sashidhar Ganiga <sganig@codeaurora.org>
Date: Tue, 7 Mar 2017 17:30:10 +0530
Subject: [PATCH 037/296] Fix for wakelock being held during bootup and idle
 scenario.

On bootup of device, if device is kept idle without unlocking the
screen the TCXO does not happen due to "ActivityManager-Sleep"
partial wake_lock being held.

Fix releases the wakelock irrespective of the result of code flow
in checkReadyForSleepLocked() function.
---
 .../core/java/com/android/server/am/ActivityStackSupervisor.java    | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/services/core/java/com/android/server/am/ActivityStackSupervisor.java b/services/core/java/com/android/server/am/ActivityStackSupervisor.java
index e62bd1251ef..4fef2d1b240 100644
--- a/services/core/java/com/android/server/am/ActivityStackSupervisor.java
+++ b/services/core/java/com/android/server/am/ActivityStackSupervisor.java
@@ -3173,6 +3173,9 @@ public class ActivityStackSupervisor extends ConfigurationContainer implements D
         applySleepTokensLocked(false /* applyToStacks */);
 
         checkReadyForSleepLocked(true /* allowDelay */);
+        if (mGoingToSleep.isHeld()) {
+            mGoingToSleep.release();
+        }
     }
 
     void prepareForShutdownLocked() {
@@ -3283,9 +3286,6 @@ public class ActivityStackSupervisor extends ConfigurationContainer implements D
 
         removeSleepTimeouts();
 
-        if (mGoingToSleep.isHeld()) {
-            mGoingToSleep.release();
-        }
         if (mService.mShuttingDown) {
             mService.notifyAll();
         }
-- 
2.11.0

